stacks-signer/src/v0/signer.rs

```diff
@@ -0,0 +1,318 @@
+        command: Option<RunLoopCommand>,
+    ) {
+        if let Some(command) = command {
+            warn!("{self}: Received a command: {command:?}. V0 Signers do not support commands. Ignoring...")
+        }
+    }
+}
+
+impl From<SignerConfig> for Signer {
+    fn from(signer_config: SignerConfig) -> Self {
+        let stackerdb = StackerDB::from(&signer_config);
+        debug!(
+            "Reward cycle #{} Signer #{}",
+            signer_config.reward_cycle, signer_config.signer_id,
+        );
+        let signer_db =
+            SignerDb::new(&signer_config.db_path).expect("Failed to connect to signer Db");
+
+        Self {
+            private_key: signer_config.stacks_private_key,
+            stackerdb,
+            mainnet: signer_config.mainnet,
+            signer_id: signer_config.signer_id,
+            signer_addresses: signer_config
+                .signer_entries
+                .signer_ids
+                .into_keys()
+                .collect(),
+            signer_slot_ids: signer_config.signer_slot_ids.clone(),
+            reward_cycle: signer_config.reward_cycle,
+            signer_db,
+        }
+    }
+}
+
+impl Signer {
+    /// Determine this signers response to a proposed block
+    /// Returns a BlockResponse if we have already validated the block
+    /// Returns None otherwise
+    fn determine_response(&self, block_info: &mut BlockInfo) -> Option<BlockResponse> {
+        let valid = block_info.valid?;
+        let response = if valid {
+            debug!("{self}: Accepting block {}", block_info.block.block_id());
+            let signature = self
+                .private_key
+                .sign(&block_info.signer_signature_hash().0)
+                .expect("Failed to sign block");
+            BlockResponse::accepted(block_info.signer_signature_hash(), signature)
+        } else {
+            debug!("{self}: Rejecting block {}", block_info.block.block_id());
+            BlockResponse::rejected(
+                block_info.signer_signature_hash(),
+                RejectCode::RejectedInPriorRound,
+            )
+        };
+        Some(response)
+    }
+
+    /// Handle block proposal messages submitted to signers stackerdb
+    fn handle_block_proposal(
+        &mut self,
+        stacks_client: &StacksClient,
+        block_proposal: &BlockProposal,
+    ) {
+        debug!("{self}: Received a block proposal: {block_proposal:?}");
+        if block_proposal.reward_cycle != self.reward_cycle {
+            // We are not signing for this reward cycle. Reject the block
+            warn!(
+                "{self}: Received a block proposal for a different reward cycle. Ignore it.";
+                "requested_reward_cycle" => block_proposal.reward_cycle,
+            );
+            return;
+        }
+        // TODO: could add a check to ignore an old burn block height if we know its oudated. Would require us to store the burn block height we last saw on the side.
```

Comment:
I think you'll need to do this at some point, but for a different reason:  the signer needs to be able to determine whether or not the block they're about to sign would conflict with an already-signed Stacks block, and if so, determine if that's allowed.